src/components/ProductDisplay/productdisplay.component.js

```diff
@@ -1,41 +1,61 @@
 import React, { Component } from 'react';
-import data from '../data'
-import Product from '../Product/product.component'
-
+import Product from '../Product/product.component';
+import { withRouter } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { getProductsRequest } from '../../actions/productsActions';
+import ReactLoading from 'react-loading';
+import "./productdisplay.css"
 class ProductDisplay extends Component {
 
     constructor(props) {
         super(props)
         this.state = {
-            pdts: data,
-        }
+            page: 0,
+            limit: 4
+        };
+    }
+
+    componentDidMount = () => {
+        const { page, limit } = this.state;
+        this.props.getProducts({ page: page, limit: limit });
+        this.setState({page: this.state.page+1})
     }
 
-    addToCart = (event) => {
-        if(this.props.isAuthenticated){
-            console.log('add to cart' + event.target.value);
-            this.props.incrementCartCount();
-        }
-        else{
-            alert('please Login to add to cart');
-        }
+    loadMoreitems = (page, limit) => {
```

Comment:
can be called as `loadMoreItems`, check naming convention standards throughout the codebase to keep consistent.
Minor Change - low priority